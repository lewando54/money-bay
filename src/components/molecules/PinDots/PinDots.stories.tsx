/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import PinDots from './PinDots'

const meta = {
	component: PinDots,
	argTypes: {
        // To write
	}
} satisfies Meta<typeof PinDots>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
	args: {
        // To write
	}
}
