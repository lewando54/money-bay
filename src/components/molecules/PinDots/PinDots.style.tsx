import { StyleSheet } from 'react-native'

import { globalColors } from '@styling/GlobalTheme.style'
export default StyleSheet.create({
    container: {
        flexDirection: 'row',
        gap: 12
    },
    base: {
        backgroundColor: globalColors.MAIN_DARK,
        width: 8,
        height: 8,
        borderRadius: 100
    },
    filled: {
        opacity: 1
    },
    notFilled: {
        opacity: 0.2
    },
})
