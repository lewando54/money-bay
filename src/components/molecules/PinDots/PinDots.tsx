import React from 'react'
import pinDotsStyle from './PinDots.style'
import { View } from 'react-native'
import Animated, { SharedValue, useAnimatedStyle } from 'react-native-reanimated'

interface IPinDotsProps {
    maxLength?: number,
    currLength?: number,
    x?: SharedValue<number>
    testID?: string
}

export default function PinDots({
    maxLength=4,
    currLength=0,
    x,
    testID
}: IPinDotsProps){
    const shakeAnimationStyle = x && useAnimatedStyle(() => ({
        transform: [{translateX: x.value}]
    }))

    const Dots = Array(maxLength).fill(0).map((val, index) => {
        if(index < currLength){
            return <View key={index} style={[pinDotsStyle.base, pinDotsStyle.filled]}></View>
        }
        else {
            return <View key={index} style={[pinDotsStyle.base, pinDotsStyle.notFilled]}></View>
        }
    })

    return (
        <Animated.View testID={testID} style={[pinDotsStyle.container, shakeAnimationStyle]}>
            {Dots}
        </Animated.View>
    )
}
