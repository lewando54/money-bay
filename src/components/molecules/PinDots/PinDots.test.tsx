import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react-native'
import PinDots from './PinDots'
import { useSharedValue } from 'react-native-reanimated'
describe('PinDots', () => {
    jest.useFakeTimers()
    it('should render correctly', () => {
        const tree = renderer.create(<PinDots />)
        expect(tree).toMatchSnapshot()
    })
})
