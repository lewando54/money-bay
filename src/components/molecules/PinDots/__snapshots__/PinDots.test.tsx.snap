// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinDots should render correctly 1`] = `
<View
  animatedStyle={
    {
      "value": {},
    }
  }
  collapsable={false}
  style={
    {
      "flexDirection": "row",
      "gap": 12,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#1B1D4D",
          "borderRadius": 100,
          "height": 8,
          "width": 8,
        },
        {
          "opacity": 0.2,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "#1B1D4D",
          "borderRadius": 100,
          "height": 8,
          "width": 8,
        },
        {
          "opacity": 0.2,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "#1B1D4D",
          "borderRadius": 100,
          "height": 8,
          "width": 8,
        },
        {
          "opacity": 0.2,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "backgroundColor": "#1B1D4D",
          "borderRadius": 100,
          "height": 8,
          "width": 8,
        },
        {
          "opacity": 0.2,
        },
      ]
    }
  />
</View>
`;
